kuma/scrape/sources/base.py

```diff
@@ -201,3 +202,48 @@ def gather(self, requester, storage):
     def load_and_validate_existing(self, storage):
         """Default: Can not load existing data from storage."""
         return False, []
+
+
+class DocumentBaseSource(Source):
+    """Shared functionality for MDN Document sources."""
+
+    PARAM_NAME = 'path'
+
+    re_path = re.compile(r"/(?P<locale>[^/]+)/docs/(?P<slug>.*)")
+
+    # Standard options for Document-based sources
+    STANDARD_DOC_OPTIONS = {
+        'force': ('bool', False),         # Update existing Document records
+        'depth': ('int_all', 0),          # Scrape the topic tree to this depth
+        'revisions': ('int', 1),          # Scrape this many past revisions
```

Comment:
Yes, and I think that was done because scrapers that ignored ``robots.txt`` were following the "all" links.  One way forward would be to enable ``?limit=all`` for anon callers, just not add the link.